cassovary-core/src/main/scala/com/twitter/cassovary/util/SortedSeq.scala

```diff
@@ -0,0 +1,114 @@
+package com.twitter.cassovary.util
+
+import java.util
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * Sequence of sorted elements.
+ * 
+ * Assumes each element may appear in the Seq at most once.
+ */
+trait StrictlySortedSeqLike[T] extends mutable.IndexedSeq[T] {
```

Comment:
I see what you mean. Will change it.

Btw, checked speed:
```
import com.twitter.cassovary.util.SortedArrayWrapper

import scala.collection.immutable.SortedSet

def time(a: => Unit): Unit = {
  val t1 = System.currentTimeMillis()
  a
  val t2 = System.currentTimeMillis()
  println("time: " + (t2 - t1))
}

val arr1: Array[Int] = (1 to 1000000).toArray

val arr2: Array[Int] = Range(1, 2000000, 2).toArray

val saw1 = SortedArrayWrapper(arr1)

val saw2 = SortedArrayWrapper(arr2)

time(saw1.intersectSorted(saw2))
val ss1: Set[Int] = SortedSet[Int]() ++ arr1
val ss2 = SortedSet[Int]() ++ arr2
time(ss1.intersect(ss2))
```
SortedArrayWrapper is consistently few orders of magnitude faster.